cmd/tendermint/commands/root.go

```diff
@@ -21,19 +21,54 @@ func init() {
 	RootCmd.PersistentFlags().String("log_level", config.LogLevel, "Log level")
 }
 
+var globalScope = "global"
+
+// excusedFlags contains flags that not set inside
+// the struct but those that are used
+var excusedFlagsMap = map[string]map[string]bool{
+	globalScope: {
+		"root":   true,
+		"help":   true,
+		"trace":  true,
+		"verify": true,
+	},
+}
+
 // ParseConfig retrieves the default environment configuration,
 // sets up the Tendermint root and ensures that the root exists
 func ParseConfig() (*cfg.Config, error) {
-	conf := cfg.DefaultConfig()
-	err := viper.Unmarshal(conf)
+	conf, err := handleConfigVerification(cfg.DefaultConfig())
 	if err != nil {
 		return nil, err
 	}
+	if err := viper.Unmarshal(conf); err != nil {
+		return nil, err
+	}
 	conf.SetRoot(conf.RootDir)
 	cfg.EnsureRoot(conf.RootDir)
 	return conf, err
 }
 
+const verifyFlag = "verify"
+
+// handleConfigVerification firstly checks if the flag `--verify` was set
+// and if not, returns the passed in config unfiltered and unchecked
+// otherwise it tallies up the values in the config file with the fields
+// of the config struct itself.
+// See https://github.com/tendermint/tendermint/issues/628.
+func handleConfigVerification(conf *cfg.Config) (*cfg.Config, error) {
+	// TODO: Move this step into viper and push up stream.
+	if !viper.GetBool(cli.VerifyFlag) {
+		return conf, nil
+	}
+	tagName := "mapstructure"
```

Comment:
why create a var? and not inline the string

Reply:
It was just an oversight. Thanks @melekes, yap I can inline that.